### PR TITLE
[macos][network][info] Improvements to DNS determination

### PR DESCRIPTION
**Pull request by enen92**
## Description
This PR brings a few improvements to `GetDNSServers` for mac. As far as I could search, using scutil is still the best way of getting those values.
The issue is that we currently don't filter per unique values, so if you have only a single DNS configured the command returns:
```
➜  xbmc git:(apple_dns) scutil --dns | grep "nameserver" | tail -n2
  nameserver[0] : 1.1.1.1
  nameserver[0] : 1.1.1.1
```

Furthermore, we sleep blindingly for 1 second to allow for the buffer of popen() to fill. I moved it to pselect with a timeout of a second instead.
Another commit added to simply display "None" instead of an empty string.

## Motivation and context
Make the code faster. Fixup inconsistencies for single DNS configuration.

## How has this been tested?
Runtime tested. Single DNS server and multiple DNS Servers.
In my case I only have 1 DNS configured (1.1.1.1), so Kodi reported this:
image

**Now:**
image

## What is the effect on users?
Faster detection of the DNS entries (important since this is exposed to the info interface), fix reporting for single DNS server.

## Screenshots (if appropriate):
See above

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp [hunk 1]
```diff
@@ -180,27 +181,45 @@ std::vector<std::string> CNetworkOsx::GetNameServers()
 {
   std::vector<std::string> result;
 
-  FILE* pipe = popen("scutil --dns | grep \"nameserver\" | tail -n2", "r");
-  usleep(100000);
+  FILE* pipe = popen("scutil --dns | grep \"nameserver\" | uniq | tail -n2", "r");
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/network/NetworkOsx.cpp:184]: probably a micro-optimization, but we could also use `grep -F nameserver`

also, why only last 2 are picked? In my case I'll be seeing only IPv6 servers:

```
  nameserver[0] : 1.1.1.1
  nameserver[1] : 1.0.0.1
  nameserver[2] : 2606:4700:4700::1111
  nameserver[3] : 2606:4700:4700::1001
```

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp [hunk 2]
```diff
@@ -180,27 +181,45 @@ std::vector<std::string> CNetworkOsx::GetNameServers()
...
+    int fileno = ::fileno(pipe);
+    fd_set read_fds;
+    FD_ZERO(&read_fds);
+    FD_SET(fileno, &read_fds);
+
+    struct timeval timeout;
+    timeout.tv_sec = 1; // Set the timeout to 1 second
+    timeout.tv_usec = 0;
+
+    int ready = select(fileno + 1, &read_fds, nullptr, nullptr, &timeout);
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/network/NetworkOsx.cpp:197]: afaik `poll` is preferred over `select`, also doesn't require using those FD_ macros

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp [hunk 3]
```diff
@@ -180,27 +181,45 @@ std::vector<std::string> CNetworkOsx::GetNameServers()
...
-      tmpStr = StringUtils::Split(buffer, "\n");
-      for (unsigned int i = 0; i < tmpStr.size(); i++)
+      char buffer[256] = {'\0'};
+      if (fgets(buffer, sizeof(buffer), pipe))
       {
         // result looks like this - > '  nameserver[0] : 192.168.1.1'
-        // 2 blank spaces + 13 in 'nameserver[0]' + blank + ':' + blank == 18 :)
-        if (tmpStr[i].length() >= 18)
-          result.push_back(tmpStr[i].substr(18));
+        // 2 blank spaces + 13 in 'nameserver[0]' + blank + ':' + blank == 18 :
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/network/NetworkOsx.cpp:205]: looks like you destroyed original smiley :)

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp [hunk 4]
```diff
@@ -180,27 +181,45 @@ std::vector<std::string> CNetworkOsx::GetNameServers()
...
+        std::vector<std::string> tmpStr = StringUtils::Split(buffer, "\n");
+        for (const std::string& str : tmpStr)
+        {
+          if (str.length() >= 18)
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/network/NetworkOsx.cpp:209]: I think this magic number deserves a constant, at least in the local scope

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp (hunk 1 above)
Comment (enen92) [reply xbmc/platform/darwin/osx/network/NetworkOsx.cpp:184]: Yeah we have two infos `NETWORK_DNS1_ADDRESS` and `NETWORK_DNS2_ADDRESS` which are populated to the first and second configured DNS servers.
Added the `-F` option although I doubt it has any noticeable difference.

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp (hunk 3 above)
Comment (enen92) [reply xbmc/platform/darwin/osx/network/NetworkOsx.cpp]: Line is now happy again :)

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp (hunk 2 above)
Comment (enen92) [reply xbmc/platform/darwin/osx/network/NetworkOsx.cpp]: Ok - moved to poll. I agree code is now a bit cleaner.

**Review by kambala-decapitator — APPROVED** [review]

**Comment by enen92** [comment]
Jenkins build and merge

**Comment by enen92** [comment]
Jenkins build and merge

**Pull request merged**